src/hypothesis/searchstrategy/strategies.py

```diff
@@ -139,6 +141,32 @@ def example(self, random=None):
         This method is part of the public API.
 
         """
+        location = current_location()
+        if location != Location.TOP_LEVEL:
+            if location == Location.INSIDE_DRAW:
+                note_deprecation(
+                    'Using example() inside a strategy definition is a bad '
```

Comment:
IMO it's worth explaining *why* this a bad idea (tests will not minimise or be reproducible), and point to strategies by fully-qualified name rather than a docs URL.

Reply:
I feel like if I listed all the things that will go wrong from doing this then the message would get rather unwieldy. Fundamentally the real reason it's a bad idea is because that's not what `example()` is for.

Reply:
I think the docs URLs are worth including, especially because most of those URLs aren't docstring references.